### DISCUSSION TITLE
Use Global Environment Variables in with clause of workflow

### DISCUSSION THREAD

**Discussion by kellervater** [community/community#25808] [Actions]
<p>I’m currently writing an action that can execute ansible playbooks.<br>
My problems is the extra-vars section.</p>
<p>Due to the fact that it’s not possible to pass arrays into the dockerfile I need to do a workaround.<br>
Mine is to pass a String like this:</p>
<pre><code>with:
  ...
  extraVars: 'my_first_extra=${{ GITHUB_ACTOR }} my_second_one=${{ GITHUB_SHA }}'
</code></pre>
<p>This doesn’t work out for me because the workflow is complaining like this:</p>
<pre><code>### ERRORED 11:48:48Z

- Your workflow file was invalid: The pipeline is not valid. .github/workflows/main.yml (Line: 23, Col: 22): Unrecognized named-value: 'GITHUB_ACTOR'. Located at position 1 within expression: GITHUB_ACTOR
</code></pre>
<p>What am I doing wrong? How can i use global environment variables as “with” variable?</p>
<p>I know that it’s possible to use secrets like <code>secrets.MY_SECRET````&amp;nbsp;   Is there a clue how to do that with global environment variables like </code>env.GITHUB_SHA```?</p>

**Comment by kellervater** [comment, marked as answer]
<p>Embarassing…<br>
A little further reading brought me to the solution which one can find here:<br>
<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions</a></p>
<p>It’s almost the same as with input variables which convert from ‘whateverMyVariable-NameIs’ to ‘INPUT_WHATEVERMYVARIABLE_NAMEIS’.<br>
The same applies here: github.sha becomes GITHUB_SHA</p>
<pre><code>with:
  ...
  extraVars: 'my_first_extra=${{ github.actor }} my_second_one=${{ github.sha }}'
</code></pre>

**Reply by PMorganCA** [reply]
you say `I know that it’s possible to use secrets` but I have tried both `secrets.secret_name` and `env.variable_name` in a `with:` clause and I get an error from the workflow like,
for example 
`with:
      target-env: ${{ env.target_env }}`

error:
`Unrecognized named-value: 'env'`

the `gitHub` context works, the `secret` and `env` contexts do not work for me.

**Comment by AndreaGriffiths11** [comment]
<p>We appreciate you coming back and sharing <a class="mention" href="/ppoetz">@ppoetz</a>, this post might help another user in the future. In the meantime, we’re looking forward to seeing you around!</p>
